src/goTest.ts

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+import cp = require('child_process');
+import path = require('path');
+import vscode = require('vscode');
+import util = require('util');
+import { getGoRuntimePath } from './goPath'
+
+/**
+* Executes the unit test at the primary cursor using `go test`. Output
+* is sent to the 'Go' channel.
+* 
+* TODO: go test returns filenames with no path information for failures,
+* so output doesn't produce navigable line references.
+*/
+export function testAtCursor() {
+	var editor = vscode.window.activeTextEditor;
+	// TODO: This command should be returning a SymbolInformation[] but is instead
+	// returning an object with slightly different fields which must be adapted.
+	vscode.commands.executeCommand<any[]>('vscode.executeDocumentSymbolProvider', editor.document.uri).then(res => {
+		var testFunction: string;
+		// Find any test function containing the cursor.
+		res.forEach(obj => {
+			var sym = newSymbolInformation(obj);
+			// Skip anything that's not a test function.
+			if (sym.kind != vscode.SymbolKind.Function) return;
+			if (!/Test.*/.exec(sym.name)) return;
+			// Determine if the function contains the primary cursor.
+			var selection = editor.selection;
+			if (selection && sym.location.range.contains(selection.start)) {
+				testFunction = sym.name;
+				return;
+			}
+		});
+		if (!testFunction) {
+			vscode.window.setStatusBarMessage('No test function found at cursor.', 5000);
+			return;
+		}
+		// Run the test and present the output in a channel.
+		// TODO: The timeout should probably be configurable somehow. An input
+		// popup seemed intrusive to ask every time.
```

Comment:
You could add a configuration setting in `package.json` to let users configure this.